### PR TITLE
Fix issues for email notifier 

### PR DESCRIPTION
**Pull request by zionwu**
Related issues:
https://github.com/rancher/rancher/issues/12672
https://github.com/rancher/rancher/issues/12655
https://github.com/rancher/rancher/issues/12654

Dependent Types PR: https://github.com/rancher/types/pull/371

### CODE REVIEW

**Comment by aiwantaozi** [comment]
LGTM, need to pay attention when the backend merged, the test function will fail before the UI be merged.

File: pkg/api/customization/alert/notifier_action.go [hunk 1]
```diff
@@ -100,11 +100,18 @@ func (h *Handler) testNotifier(actionName string, action *types.Action, apiConte
 			host, ok := smtpConfig["host"].(string)
 			if ok {
 				port, _ := smtpConfig["port"].(json.Number).Int64()
-				password := smtpConfig["password"].(string)
+				password, ok := smtpConfig["password"].(string)
```
Comment (alena1108) [inline pkg/api/customization/alert/notifier_action.go]: I'd suggest you use `convert` package to do strings conversion. Its string() method would do automatic defaulting to empty string as well, so the line 104 is no longer needed. Same for username and the sender

File: pkg/api/customization/alert/notifier_action.go [hunk 2]
```diff
@@ -246,54 +253,66 @@ func testSlack(url, channel, msg string) error {
...
 	var c *smtp.Client
 	smartHost := host + ":" + strconv.Itoa(port)
 
 	timeout := 15 * time.Second
-	if requireTLS {
+	if port == 465 {
 		conn, err := tls.DialWithDialer(&net.Dialer{Timeout: timeout}, "tcp", smartHost, &tls.Config{ServerName: host})
 		if err != nil {
-			return err
+			return fmt.Errorf("Failed to connect smtp server: %s", err)
```
Comment (alena1108) [inline pkg/api/customization/alert/notifier_action.go]: the error would be %v, not %s

File: pkg/api/customization/alert/notifier_action.go (hunk 1 above)
Comment (zionwu) [reply pkg/api/customization/alert/notifier_action.go]: Sure

File: pkg/api/customization/alert/notifier_action.go (hunk 2 above)
Comment (zionwu) [reply pkg/api/customization/alert/notifier_action.go]: ok

**Pull request merged**